pkg/pillar/zedUpload/azureutil/azure.go

```
@@ -242,3 +244,91 @@ func GetAzureBlobMetaData(accountName, accountKey, containerName, remoteFile str
 	stringHex := hex.EncodeToString(decodedString)
 	return blob.Properties.ContentLength, stringHex, nil
 }
+
+// GenerateBlobSasURI is used to generate the URI which can be used to access the blob until the the URI expries
+func GenerateBlobSasURI(accountName, accountKey, containerName, remoteFile string, httpClient *http.Client) (string, error) {
+	c, err := NewClient(accountName, accountKey, httpClient)
+	if err != nil {
+		return "", err
+	}
+	blobClient := c.GetBlobService()
+	container := blobClient.GetContainerReference(containerName)
+	containerExists, _ := container.Exists()
+	if !containerExists {
+		return "", err
+	}
+	blob := container.GetBlobReference(remoteFile)
+	exists, err := blob.Exists()
+	if err != nil {
+		return "", err
+	}
+	if !exists {
+		return "", fmt.Errorf("blob doesn't exist")
+	}
+	options := storage.BlobSASOptions{}
+	options.Read = true
+	options.Start = time.Now()
+	options.Expiry = time.Now().Add(time.Hour * 2)
+	options.UseHTTPS = true
+
+	sasURI, err := blob.GetSASURI(options)
+	if err != nil {
+		return "", err
+	}
+	return sasURI, nil
+}
+
+// UploadPartByChunk is used to upload the given chunk of data into the block blob
+func UploadPartByChunk(accountName, accountKey, containerName, remoteFile, partID string, httpClient *http.Client, chunk []byte) error {
+	c, err := NewClient(accountName, accountKey, httpClient)
+	if err != nil {
+		return err
+	}
+	blobClient := c.GetBlobService()
+	container := blobClient.GetContainerReference(containerName)
+	containerExists, _ := container.Exists()
+	if !containerExists {
+		err = container.Create(nil)
+		if err != nil {
+			return err
+		}
+	}
+	blob := container.GetBlobReference(remoteFile)
+	exists, _ := blob.Exists()
+	if !exists {
+		err = blob.CreateBlockBlob(nil)
+		if err != nil {
+			fmt.Println("erorr while creating block blob", err)
```

Please return an error instead of printing it.